### PR TITLE
boards: google: add google_icetower board

### PR DESCRIPTION
**Pull request by niedzwiecki-dawid**
Introduce Google Icetower Development Board.
Icetower is a board created by Google for fingerprint-related functionality development.

### CODE REVIEW

**Review by kartben — CHANGES_REQUESTED** [review]

File: boards/google/icetower/doc/index.rst [hunk 1]
```diff
@@ -0,0 +1,48 @@
+.. _google_icetower_board:
+
+Google Icetower Development Board
+###################################
```
[the remark this answers is not in the document]
Comment (niedzwiecki-dawid) [reply boards/google/icetower/doc/index.rst]: Done.

**Review by fabiobaltieri — DISMISSED** [review]

File: boards/google/icetower/board.yml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+board:
+  name: google_icetower
+  full_name: Google Icetower Development Board
```
Comment (kartben) [inline boards/google/icetower/board.yml:3]:
```suggestion
  full_name: Icetower Development Board
```

Comment (kartben) [reply boards/google/icetower/board.yml:3]: see https://builds.zephyrproject.io/zephyr/pr/83034/docs/boards/index.html#vendor=google -- vendor name is redundant in the full name

Comment (niedzwiecki-dawid) [reply boards/google/icetower/board.yml]: Done.

**Review by kartben — APPROVED** [review]

**Review by fabiobaltieri — APPROVED** [review]

**Review by duda-patryk — APPROVED** [review]

**Review by nordicjm — APPROVED** [review]

**Review by keith-zephyr — APPROVED** [review]

**Pull request merged**